src/Mod/Fem/Gui/ViewProviderFemConstraintOnBoundary.cpp

```diff
@@ -56,16 +56,13 @@ void ViewProviderFemConstraintOnBoundary::highlightReferences(const bool on)
     const auto& subSets = pcConstraint->References.getSubListValues();
 
     for (auto& subSet : subSets) {
-        // if somehow the subnames are empty, we have nothing to do
-        if (subSet.second.empty())
-            continue;
-
         Part::Feature* base = static_cast<Part::Feature*>(subSet.first);
```

Comment:
This static_cast is problematic because there is no guarantee that the objects in the References property are all Part::Feature. A dynamic_cast and a null check is needed here.

Reply:
Check added.